Uplift project

yads opened the pull request:
various updates to modernize project

* use package-lock.json
* update various packages and add peerDependencies
* update tests to use headless browsers and use testem harness
* add linting
* update tests to create elements inside the document and fix setup to use beforeEach constructs

I've based it on the rendered text PR as well so this has those changes as well, plus adds tests for the rendered option

nathanboktae commented on test/tests.js:
```diff
@@ -224,7 +234,9 @@ describe('DOM assertions', function() {
-      var subject = parse('<div> foo </div>')
+      beforeEach(function() {
+        subject = parse('<div> foo </div>')
+      });
```
I like these changes, thanks.

nathanboktae commented on test/tests.js:
```diff
@@ -304,13 +335,13 @@ describe('DOM assertions', function() {
           (function() {
             subject.should.not.have.text(fullText)
-          }).should.fail("expected span, div, span, p, span not to have text '" + fullText + "'")
+          }).should.fail("expected span, div, span, p, span not to have text '" + fullText + "'") // eslint-disable-line quotes
```
Rather let's just turn off `eslint-disable-line` for now.

nathanboktae commented on test/tests.js:
```diff
@@ -594,7 +633,9 @@ describe('DOM assertions', function() {
-      var subject = parse('<div><span class="blurb">example text</span><p>lorem ipsum</p></div>')
+      beforeEach(function() {
+        subject = parse('<div><span class="blurb">example text</span><p>lorem ipsum</p></div>')
+      });
```
Since you built this PR off of the `rendered` text additions, how about add tests here now?

nathanboktae reviewed (changes requested):
Wow this is awesome! thanks for the work! Some minor linting comments and tests for the `rendered` case and I'll be very happy to take this PR.

yads commented:
I updated the rendered code based on your suggestion in that PR and added more tests for it. Updated based off your other suggestions. Decided not to putz around with getting the old test structure to work. So we can likely close the other PR once this is merged in.

nathanboktae commented:
Thanks a ton for this! It's been much needed to get off of `phantomjs`.

yads commented:
Woot!
